bindings/node/test/clientEncryption.test.js

```diff
@@ -815,4 +829,115 @@ describe('ClientEncryption', function () {
   it('should provide the libmongocrypt version', function () {
     expect(ClientEncryption.libmongocryptVersion).to.be.a('string');
   });
+
+  describe('createEncryptedCollection()', () => {
+    /** @type {InstanceType<ClientEncryption>} */
+    const client = new MockClient();
+    let clientEncryption;
+    let db;
+    const collectionName = 'secure';
+
+    beforeEach(async function () {
+      clientEncryption = new ClientEncryption(client, {
+        keyVaultNamespace: 'client.encryption',
+        kmsProviders: { local: { key: Buffer.alloc(96, 0) } }
+      });
+
+      db = client.db('createEncryptedCollectionDb');
+      await db.dropDatabase().catch(() => null);
+    });
+
+    afterEach(async () => {
+      sinon.restore();
+      await db.dropDatabase().catch(() => null);
+    });
+
+    it('throws if options are omitted', async () => {
+      const error = await clientEncryption
+        .createEncryptedCollection(db, collectionName)
+        .catch(error => error);
```

Comment:
We're missing a test for the case when `encryptedFieldsMap` is not an array (we guard against it, we should test it).  Also, should we throw in that case instead of just not entering the if-statement?

Reply:
`encryptedFields.fields`? The spec has only an if stmt for this, I think we shouldn't add more validation than is needed for the helper. Added a test for this, and for when an item is not a document

Reply:
I think we should revisit this.  The spec says to silently continue but if we do, a non-array `encryptedFields` will silently succeed (this method will not reject), the `createCollection` call succeeds but no data keys were generated.

This pseudocode will fail, probably surprisingly, and require the user to manually delete the created collections:
```typescript
const options = { encryptedFields: {} }; // non-array
const newEncryptedFields = clientEncryption.createEncryptedCollection(.... , options);

const client = await new MongoClient(..., { encryptedFieldsMap: newEncryptedFields }).connect() // throws
```

There have been multiple modifications to the spec - I think we should consider this as well.  Thoughts?  @nbbeeken for visibility

Reply:
I leave that up to the spec owners, so let's circle back when there's word.

Something to consider, the more validation we pack into this function the more coordination it needs to maintain with server-side data structures, maybe there's a future where fields isn't an array and this function can continue to be invoked without a breaking a new shape.

Reply:
From slack: We filed DRIVERS-2540, and we will not do that work here.